File: CS/WindowsApplication53/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraCharts;
using DevExpress.XtraPivotGrid;

namespace WindowsApplication53
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void Form1_Load(object sender, EventArgs e)
        {
            PopulateTable();
            pivotGridControl1.RefreshData();
            pivotGridControl1.BestFit();
        }
        private void PopulateTable()
        {
            DataTable myTable = dataSet1.Tables["Data"];
            myTable.Rows.Add(new object[] { "Aaa", DateTime.Today, 7 });
            myTable.Rows.Add(new object[] { "Aaa", DateTime.Today.AddDays(1), 4 });
            myTable.Rows.Add(new object[] { "Bbb", DateTime.Today, 12 });
            myTable.Rows.Add(new object[] { "Bbb", DateTime.Today.AddDays(1), 14 });
            myTable.Rows.Add(new object[] { "Ccc", DateTime.Today, 11 });
            myTable.Rows.Add(new object[] { "Ccc", DateTime.Today.AddDays(1), 10 });

            myTable.Rows.Add(new object[] { "Aaa", DateTime.Today.AddDays(3), 4 });
            myTable.Rows.Add(new object[] { "Aaa", DateTime.Today.AddDays(2), 2 });
            myTable.Rows.Add(new object[] { "Bbb", DateTime.Today.AddDays(3), 3 });
            myTable.Rows.Add(new object[] { "Bbb", DateTime.Today.AddDays(2), 1 });
            myTable.Rows.Add(new object[] { "Ccc", DateTime.Today.AddDays(3), 8 });
            myTable.Rows.Add(new object[] { "Ccc", DateTime.Today.AddDays(2), 22 });
        }

        private void pivotGridControl1_CustomChartDataSourceData(object sender, DevExpress.XtraPivotGrid.PivotCustomChartDataSourceDataEventArgs e)
        {

        }


        Point hotTrackPoint = new Point(-1, -1);
        private void chartControl1_ObjectHotTracked(object sender, DevExpress.XtraCharts.HotTrackEventArgs e)
        {
            SeriesPoint point = e.AdditionalObject as SeriesPoint;
            if (point == null)
            {
                InvalidateCell(pivotGridControl1, hotTrackPoint);
                hotTrackPoint = new Point(-1, -1);
            }
            else
            {
                PivotChartDataSourceRowItem coordinates = point.Tag as PivotChartDataSourceRowItem;

                InvalidateCell(pivotGridControl1, hotTrackPoint);
                hotTrackPoint.X = coordinates.CellX ;
                hotTrackPoint.Y = coordinates.CellY ;
                InvalidateCell(pivotGridControl1, hotTrackPoint);
            }
        }

        private void InvalidateCell(DevExpress.XtraPivotGrid.PivotGridControl pivot, Point cell)
        {
            if (cell != new Point(-1, -1))
                pivot.Invalidate(pivot.Cells.GetCellInfo(cell.X, cell.Y).Bounds);
        }

        private void pivotGridControl1_CustomDrawCell(object sender, DevExpress.XtraPivotGrid.PivotCustomDrawCellEventArgs e)
        {

        }

        private void pivotGridControl1_CustomAppearance(object sender, DevExpress.XtraPivotGrid.PivotCustomAppearanceEventArgs e)
        {
            if (hotTrackPoint == new Point(-1, -1)) return;
            if (e.RowIndex == hotTrackPoint.Y && e.ColumnIndex == hotTrackPoint.X)
                e.Appearance.BackColor = Color.LightGreen;
        }

    }
}

File: VB/WindowsApplication53/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraCharts
Imports DevExpress.XtraPivotGrid

Namespace WindowsApplication53

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            PopulateTable()
            pivotGridControl1.RefreshData()
            pivotGridControl1.BestFit()
        End Sub

        Private Sub PopulateTable()
            Dim myTable As DataTable = dataSet1.Tables("Data")
            myTable.Rows.Add(New Object() {"Aaa", Date.Today, 7})
            myTable.Rows.Add(New Object() {"Aaa", Date.Today.AddDays(1), 4})
            myTable.Rows.Add(New Object() {"Bbb", Date.Today, 12})
            myTable.Rows.Add(New Object() {"Bbb", Date.Today.AddDays(1), 14})
            myTable.Rows.Add(New Object() {"Ccc", Date.Today, 11})
            myTable.Rows.Add(New Object() {"Ccc", Date.Today.AddDays(1), 10})
            myTable.Rows.Add(New Object() {"Aaa", Date.Today.AddDays(3), 4})
            myTable.Rows.Add(New Object() {"Aaa", Date.Today.AddDays(2), 2})
            myTable.Rows.Add(New Object() {"Bbb", Date.Today.AddDays(3), 3})
            myTable.Rows.Add(New Object() {"Bbb", Date.Today.AddDays(2), 1})
            myTable.Rows.Add(New Object() {"Ccc", Date.Today.AddDays(3), 8})
            myTable.Rows.Add(New Object() {"Ccc", Date.Today.AddDays(2), 22})
        End Sub

        Private hotTrackPoint As Point = New Point(-1, -1)

        Private Sub chartControl1_ObjectHotTracked(ByVal sender As Object, ByVal e As HotTrackEventArgs)
            Dim point As SeriesPoint = TryCast(e.AdditionalObject, SeriesPoint)
            If point Is Nothing Then
                InvalidateCell(pivotGridControl1, hotTrackPoint)
                hotTrackPoint = New Point(-1, -1)
            Else
                Dim coordinates As PivotChartDataSourceRowItem = TryCast(point.Tag, PivotChartDataSourceRowItem)
                InvalidateCell(pivotGridControl1, hotTrackPoint)
                hotTrackPoint.X = coordinates.CellX
                hotTrackPoint.Y = coordinates.CellY
                InvalidateCell(pivotGridControl1, hotTrackPoint)
            End If
        End Sub

        Private Sub InvalidateCell(ByVal pivot As PivotGridControl, ByVal cell As Point)
            If cell <> New Point(-1, -1) Then pivot.Invalidate(pivot.Cells.GetCellInfo(cell.X, cell.Y).Bounds)
        End Sub

        Private Sub pivotGridControl1_CustomAppearance(ByVal sender As Object, ByVal e As PivotCustomAppearanceEventArgs)
            If hotTrackPoint = New Point(-1, -1) Then Return
            If e.RowIndex = hotTrackPoint.Y AndAlso e.ColumnIndex = hotTrackPoint.X Then e.Appearance.BackColor = Color.LightGreen
        End Sub
    End Class
End Namespace
